### DISCUSSION TITLE
Why Next (npm run dev) faster than React (npm start)

### DISCUSSION THREAD

**Discussion by thanhtutzaw** [vercel/next.js#42472] [Help] [answered]
Why Next (npm run dev) faster than React (npm start)

**Comment by icyJoseph** [comment, marked as answer]
Various reasons:

> Assuming React means create-react-app. React is a library to make UI not an application framework, or scaffolding package.

- `next dev`, uses a rust based compiler called [swc](https://swc.rs/), vite uses esbuild, create-react-app (I guess you mean that when you say React) is just JS out of the box, AFAIK.
- `next dev` only compiles resources you request, running `next dev` and doing nothing compiles only a few assets, you actually have to go to the route you wish to see to trigger on-demand compilation
- Since Next.js is route based, upon dev start, and when a page is demanded, it only needs to compile the JavaScript/TypeScript associated with that route, and not the entire monolith. 
- Webpack and a bunch of 3rd party dependencies are actually pre-compiled. If you do `npm explain webpack` or `yarn why webpack` on a Next.js project, the tool will say that there's no such dependency! They are present, but they are compiled by Next.js, so that when you run them they don't need to transpiled or whatever other process they might need.
